packages/compiler-core/src/transforms/vOnce.ts

```diff
@@ -11,10 +11,13 @@ export const transformOnce: NodeTransform = (node, context) => {
       return
     }
     seen.add(node)
+    const wasInVonce = context.inVOnce
```

Comment:
Shouldn't we just ignore nested `v-once` transforms by returning immediately at top?